mcp/src/main/java/io/modelcontextprotocol/server/transport/StreamableHttpServerTransportProvider.java

```diff
@@ -0,0 +1,765 @@
+	 * @throws IOException If an I/O error occurs
+	 */
+	@Override
+	protected void doDelete(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+
+		String requestURI = request.getRequestURI();
+		if (!requestURI.endsWith(mcpEndpoint)) {
+			response.sendError(HttpServletResponse.SC_NOT_FOUND);
+			return;
+		}
+
+		String sessionId = request.getHeader(SESSION_ID_HEADER);
+		if (sessionId == null) {
+			response.setContentType(APPLICATION_JSON);
+			response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+			response.getWriter().write(createErrorJson("Session ID missing in request header"));
+			return;
+		}
+
+		McpServerSession session = sessions.remove(sessionId);
+		if (session == null) {
+			response.setContentType(APPLICATION_JSON);
+			response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+			response.getWriter().write(createErrorJson("Session not found: " + sessionId));
+			return;
+		}
+
+		// Close the session and any associated SSE stream
+		StreamableHttpSseStream sseStream = sseStreams.remove(sessionId);
+		if (sseStream != null) {
+			sseStream.complete();
+		}
+
+		session.close();
+		logger.debug("Session terminated: {}", sessionId);
+
+		response.setStatus(HttpServletResponse.SC_OK);
+	}
+
+	/**
+	 * Gets or creates a session for the given session ID.
+	 * @param sessionId The session ID
+	 * @param createIfMissing Whether to create a new session if one doesn't exist
+	 * @return The session, or null if it doesn't exist and createIfMissing is false
+	 */
+	private McpServerSession getOrCreateSession(String sessionId, boolean createIfMissing) {
+		McpServerSession session = sessions.get(sessionId);
+		if (session == null && createIfMissing) {
+			StreamableHttpServerTransport transport = new StreamableHttpServerTransport(sessionId);
+			session = sessionFactory.create(transport);
+			sessions.put(sessionId, session);
+			logger.debug("Created new session: {}", sessionId);
+		}
+		return session;
+	}
+
+	/**
+	 * Gets or creates an SSE stream for the given session ID.
+	 * @param sessionId The session ID
+	 * @return The SSE stream
+	 */
+	private StreamableHttpSseStream getOrCreateSseStream(String sessionId) {
+		return sseStreams.computeIfAbsent(sessionId, id -> {
+			StreamableHttpSseStream stream = new StreamableHttpSseStream();
+			logger.debug("Created new SSE stream for session: {}", id);
+			return stream;
+		});
+	}
+
+	/**
+	 * Creates a JSON error response.
+	 * @param message The error message
+	 * @return The JSON error string
+	 */
+	private String createErrorJson(String message) {
+		try {
+			return objectMapper.writeValueAsString(new McpError(message));
+		}
+		catch (IOException e) {
+			logger.error("Failed to serialize error message", e);
+			return "{\"error\":\"" + message + "\"}";
+		}
+	}
+
+	/**
+	 * Implementation of McpServerTransport for Streamable HTTP sessions.
+	 */
+	private class StreamableHttpServerTransport implements McpServerTransport {
```

Comment:
I am guessing for Streamable HTTP we need a new abstraction, a `McpStreamableServerTransport` that has a different set of methods allowing opening a new stream and staying with the `McpServerTransport` in case of simple (sessionless) servers.